### DISCUSSION TITLE
Use static_cast<T&&> instead of std::forward to reduce template instantiations???

### DISCUSSION THREAD

**Discussion by stephenberry** [stephenberry/glaze#625] [Optimization]
I've seen codebases use a macro for forwarding, or just use `static_cast<T&&>(...)` to avoid the template instantiations for `std::forward` and save a bit of compile time. This might be worthwhile for Glaze.

However, the biggest boost to compilation performance will be moving to C++20 modules.

**Comment by justend29** [comment]
would removing instantiations of `std::forward` save anything even measurable?

**Reply by stephenberry** [reply]
This would have no runtime effect at all. I've just heard of this change reducing compile times by around 5%

**Reply by justend29** [reply]
Yes, the compile-times. 5% is much higher than I would ever have expected.
"forward" does convey additional expressiveness compared to static_cast.
